### ISSUE TITLE
`fastparquet.api.sorted_partitioned_columns()` does not account for `filters`

### ISSUE THREAD

**Issue by andrethrill** [dask/fastparquet#370]
### TLDR

`fastparquet.api.sorted_partitioned_columns()` currently checks the entire parquet file without considering the possibility of `filters`. 

*I'm uploading a PR in a few moments that fixes this.*

This, in turn, is useful for https://github.com/dask/dask/issues/3930.

### Imports

```python
import numpy as np
import pandas as pd
import dask.dataframe as dd
import fastparquet
from fastparquet import ParquetFile
```

### Generate dummy data. Save to a partitioned parquet file

```python
def generate_df(size, id_val=None):
    df = pd.DataFrame({'unique' : np.arange(0, size),
                       'dates' : pd.date_range('2015-01-01',
                                             periods=size,
                                             freq='1min',
                                             name='dates')})
    if id_val is not None:
        df['id']=id_val
    return df
```

```python
df_list = [generate_df(4, id_val = id_val) for id_val in ['id1','id2'] ]
df_pd = pd.concat(df_list).sort_values('dates').set_index('dates')
df_pd
```

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>unique</th>
      <th>id</th>
    </tr>
    <tr>
      <th>dates</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2015-01-01 00:00:00</th>
      <td>0</td>
      <td>id1</td>
    </tr>
    <tr>
      <th>2015-01-01 00:00:00</th>
      <td>0</td>
      <td>id2</td>
    </tr>
    <tr>
      <th>2015-01-01 00:01:00</th>
      <td>1</td>
      <td>id1</td>
    </tr>
    <tr>
      <th>2015-01-01 00:01:00</th>
      <td>1</td>
      <td>id2</td>
    </tr>
    <tr>
      <th>2015-01-01 00:02:00</th>
      <td>2</td>
      <td>id1</td>
    </tr>
    <tr>
      <th>2015-01-01 00:02:00</th>
      <td>2</td>
      <td>id2</td>
    </tr>
    <tr>
      <th>2015-01-01 00:03:00</th>
      <td>3</td>
      <td>id1</td>
    </tr>
    <tr>
      <th>2015-01-01 00:03:00</th>
      <td>3</td>
      <td>id2</td>
    </tr>
  </tbody>
</table>
</div>

```python
df = dd.from_pandas(df_pd, npartitions=2)
```

```python
# The divisions
df.divisions
```

    (Timestamp('2015-01-01 00:00:00'),
     Timestamp('2015-01-01 00:02:00'),
     Timestamp('2015-01-01 00:03:00'))

```python
# 2 partitions as expected
print('partitions: ' + str(df.npartitions))
```

    partitions: 2

```python
df.to_parquet('./data.parquet', compression='snappy',
              engine='fastparquet', partition_on=['id'])
```

### Load ParquetFile

```python
pq_f = ParquetFile('./data.parquet')
```

```python
# the group files
pq_f.group_files
```

    {0: {'id=id1/part.0.parquet'},
     1: {'id=id2/part.0.parquet'},
     2: {'id=id1/part.1.parquet'},
     3: {'id=id2/part.1.parquet'}}

### The output of sorted_partitioned_columns()

Assume I want to filter the parquet file only based on the following filters:

```python
filters = [('id', '==', 'id1')]
```

At the moment, `sorted_partitioned_columns()` does not take `filters` into consideration and, as expected, outputs an empty `dict()`:

```python
fastparquet.api.sorted_partitioned_columns(pq_f)
```

    {}

I'm soon creating a PR that will allow to take into consideration `filters` when checking for sorted columns partition-by-partition:

```python
output = fastparquet.api.sorted_partitioned_columns(pq_f, filters=filters)
output
```

    {'dates': {'min': [numpy.datetime64('2015-01-01T00:00:00.000000000'),
       numpy.datetime64('2015-01-01T00:02:00.000000000')],
      'max': [numpy.datetime64('2015-01-01T00:01:00.000000000'),
       numpy.datetime64('2015-01-01T00:03:00.000000000')]},
     'unique': {'min': [0, 2], 'max': [1, 3]}}

This, in turn, is useful for https://github.com/dask/dask/issues/3930.
